AirCasting/Utils/Averaging.swift

```diff
@@ -0,0 +1,151 @@
+    // TODO:
+    // 1. calculate average
+    // 2. remove unaveraged measurements
+    // 3. mark measurements as averaged
+    // 4. setup timer for new session
+    // 5. perform averaging of previous measurements after crossing threshold
+    // 5. get proper averaging window (for threshold)
+    
+    private var sessionEntity: SessionEntity?
+    private let measurementStreamStorage: MeasurementStreamStorage
+    private var timers: [SessionUUID : AnyCancellable] = [:]
+    private var fetchedResultsController: NSFetchedResultsController<SessionEntity>?
+    
+    init(measurementStreamStorage: MeasurementStreamStorage) {
+        self.measurementStreamStorage = measurementStreamStorage
+        super.init()
+        
+        measurementStreamStorage.accessStorage { [weak self] storage in
+            let request: NSFetchRequest<SessionEntity> = SessionEntity.fetchRequest()
+            request.predicate = NSPredicate(format: "type == %@ AND status == %i",
+                                            SessionType.mobile.rawValue,
+                                            SessionStatus.RECORDING.rawValue)
+            request.sortDescriptors = [NSSortDescriptor(key: "type", ascending: true)]
+            
+            let frc = storage.observerFor(request: request)
+            
+            try! frc.performFetch()
+            print("[🦆] Start observer")
+            frc.delegate = self
+            self?.fetchedResultsController = frc
+        }
+    }
+    
+    private func perform(sessionEntity: SessionEntity, elementsInChunk: AveragingWindow) {
+        if let uuid = sessionEntity.uuid {
+            measurementStreamStorage.accessStorage { storage in
+                
+                let session = try? storage.getExistingSessionWith(uuid)
+                print("[🦆] AVERAGING: perform()")
+                
+                session?.allStreams?.forEach { stream in
+                    var averagedMeasurements: [MeasurementEntity] = []
+                    let unaveragedMeasurements: [MeasurementEntity] = (stream.allMeasurements ?? []).filter {
+                        $0.averagingWindow == AveragingWindow.zeroWindow.rawValue
+                    }
+                    
+                    print("[🦆] Unaveraged measurements: \(unaveragedMeasurements.count)")
+                    unaveragedMeasurements.chunks(ofCount: elementsInChunk.rawValue).forEach( { measuremensInChunk in
+                        print("[🦆] B")
+                        if (measuremensInChunk.count == elementsInChunk.rawValue) {
+                            /// https://github.com/apple/swift-algorithms/blob/main/Guides/Chunked.md
+                            /// For integer types, any remainder of the division is discarded so we need to add 1 to elementsInChunk/2 to get the middle value.
+                            let idx = measuremensInChunk.index(measuremensInChunk.startIndex, offsetBy: (elementsInChunk.rawValue/2 + 1))
```

Comment:
middleIndex? middleIdx?

Reply:
reminder: explain me why do we need all that weird stuff

Reply:
I'll try to explain.
Introduction: we're using Apple's Algorithms `chunks(ofCount: )` method. This method doesn'e chunk given array into "subarrays", from documentation: "Returns a ChunksOfCountCollection<Self> view presenting the elements in chunks with count of the given count parameter."
So, after I changed the old method to apple's `chunks(ofCount: )` I was getting the error here: 
`let middleMeasurement = measuremensInChunk[middleIndex] `
that the index is out of range. It turned out, the elements of this ChunksOfCountCollection have the index of the original array (unaveragedMeasurements). That's why I use `(measuremensInChunk.startIndex, offsetBy: (averagingWindow.rawValue/2 + 1))` :) 

Not sure if it's clear. So for example: 

averagedMeasurements = [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]